### PR TITLE
Profiles support in OTLP Exporter

### PR DESCRIPTION
**Pull request by dmathieu**
#### Description

This adds profiles support to the OTLP exporters.
Note: This includes some commits from #11071 that were required in here too.

### CODE REVIEW

File: exporter/exporterhelper/profiles.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	return req.pd.SampleCount()
+}
+
+type profileExporter struct {
+	*baseExporter
+	consumerprofiles.Profiles
+}
+
+// NewProfilesExporter creates an exporterprofiles.Profiless that records observability metrics and wraps every request with a Span.
+func NewProfilesExporter(
```
Comment (mx-psi) [inline exporter/exporterhelper/profiles.go:74]: We will eventually have to move this away to a different package when we mark exporterhelper as 1.0. We can do this now on this PR or I can do it later sicne there is other experimental API

Comment (dmathieu) [reply exporter/exporterhelper/profiles.go]: Done in https://github.com/open-telemetry/opentelemetry-collector/pull/11131/commits/c14ae5aba347341a7c1f854ad4b239a96f26b38f.

**Comment by mx-psi** [comment]
@dmathieu Can you fix the merge conflicts?

**Comment by dmathieu** [comment]
Closing this PR, as everything it contains has been split into multiple smaller ones.

**Pull request closed without merging**